src/SayMore/UI/ExportMediaDlg.cs

```diff
@@ -0,0 +1,260 @@
+		}
+
+		/// ------------------------------------------------------------------------------------
+		public ExportMediaDlg(ConvertMediaDlgViewModel viewModel) : this()
+		{
+			Logger.WriteEvent("ExportMediaDlg constructor. file = {0}", viewModel.InputFile);
+			_viewModel = viewModel;
+
+			if (Settings.Default.ExportMediaDlg == null)
+			{
+				StartPosition = FormStartPosition.CenterScreen;
+				Settings.Default.ExportMediaDlg = FormSettings.Create(this);
+			}
+
+			InitializeFonts();
+
+			if (_viewModel.MediaInfo == null)
+			{
+				_labelStatus.Text = LocalizationManager.GetString("DialogBoxes.ConvertMediaDlg.InvalidMediaFile",
+					"File does not appear to be a valid media file.");
+				_labelStatus.ForeColor = Color.Red;
+			}
+			else
+			{
+				_labelOutputFileValue.Text = _viewModel.GetNewOutputFileName(true);
+				_buttonCancel.Click += delegate { _viewModel.Cancel(); };
+
+				_buttonCancel.MouseMove += delegate
+				{
+					// Not sure why both of these are necessary, but it seems to be the case.
+					_buttonCancel.UseWaitCursor = false;
+					_buttonCancel.Cursor = Cursors.Default;
+				};
+			}
+
+			_buttonDownload.Click += delegate
+			{
+				using (var dlg = new FFmpegDownloadDlg())
+					dlg.ShowDialog(this);
+				_viewModel.SetConversionStateBasedOnPresenceOfFfmpegForSayMore();
+				UpdateDisplay();
+			};
+
+			Shown += delegate
+			{
+				BeginConversion();
+			};
+
+			_labelDownloadNeeded.Tag = _labelDownloadNeeded.Text;
+			LocalizeItemDlg.StringsLocalized += HandleStringsLocalized;
+
+			Program.SuspendBackgroundProcesses();
+		}
+
+		/// ------------------------------------------------------------------------------------
+		private void HandleStringsLocalized()
+		{
+			_labelDownloadNeeded.Tag = _labelDownloadNeeded.Text;
+			UpdateDisplay();
+		}
+
+		/// ------------------------------------------------------------------------------------
+		protected override void OnLoad(EventArgs e)
+		{
+			Settings.Default.ExportMediaDlg.InitializeForm(this);
+			base.OnLoad(e);
+			UpdateDisplay();
+		}
+
+		/// ------------------------------------------------------------------------------------
+		protected override void OnClosing(CancelEventArgs e)
+		{
+			Program.ResumeBackgroundProcesses(true);
+
+			if ((_viewModel.ConversionState & ConvertMediaUIState.FinishedConverting) > 0)
+				DialogResult = DialogResult.OK;
+
+			base.OnClosing(e);
+		}
+
+		/// ------------------------------------------------------------------------------------
+		private void InitializeFonts()
+		{
+			_textBoxOutput.Font = Program.DialogFont;
+			_labelOutputFile.Font = Program.DialogFont;
+			_labelOutputFileValue.Font = FontHelper.MakeFont(Program.DialogFont, FontStyle.Bold);
+			_labelDownloadNeeded.Font = Program.DialogFont;
+			_labelStatus.Font = _labelOutputFileValue.Font;
+		}
+
+		/// ------------------------------------------------------------------------------------
+		public static string GetFactoryConvertToH263Mp4ConversionName()
```

Comment:
We don't want to duplicate these strings. The ones here are not used anyway, but by having two different instances of the same localizable string, you set up a situation where they could end up getting out of sync and L10nSharp will choose one arbitrarily.

Reply:
Combined into one.